generic/src/main/scala-3/com/tersesystems/echopraxia/plusscala/generic/Derivation.scala

```diff
@@ -0,0 +1,105 @@
+package com.tersesystems.echopraxia.plusscala.generic
+
+import magnolia1.*
+import magnolia1.CaseClass.Param
+
+import scala.deriving.Mirror
+
+import com.tersesystems.echopraxia.plusscala.api._
+import com.tersesystems.echopraxia.api.Value
+import com.tersesystems.echopraxia.api.Field
+
+/**
+ * This trait uses Magnolia to provide generic type class derivation for case classes and sealed traits. Note that you need to include the
+ * `scala-reflect` library to use it, see installation instructions.
+ *
+ * If you create a field builder that uses derivation, it will automatically log case classes where all the attributes have `ToValue` type classes in
+ * scope.
+ *
+ * {{{
+ * final case class IceCream(name: String, numCherries: Int, inCone: Boolean)
+ *
+ * final case class EntityId(raw: Int) extends AnyVal
+ * final case class Bar(underlying: String) extends AnyVal
+ * final case class Foo(bar: Bar)
+ *
+ * trait MagnoliaFieldBuilder extends FieldBuilder with AutoDerivation
+ *
+ * object MagnoliaFieldBuilder extends MagnoliaFieldBuilder
+ *
+ * object MagnoliaMain {
+ *   private val logger = LoggerFactory.getLogger.withFieldBuilder(MagnoliaFieldBuilder)
+ *
+ *   def main(args: Array[String]): Unit = {
+ *     logger.info("{}", _.keyValue("icecream", IceCream("sundae", 1, false)))
+ *     logger.info("{}", _.keyValue("entityId", EntityId(1)))
+ *     logger.info("{}", _.keyValue("foo", Foo(Bar("underlying"))))
+ *   }
+ * }
+ * }}}
+ */
+trait AutoDerivation { self: ValueTypeClasses =>
+
+  object ToValueAutoDerivation extends magnolia1.AutoDerivation[ToValue] {
```

Comment:
the magnolia api is a bit different between 1.2 and 1.3 so have to use a different implementation